be/src/util/threadpool.cpp

```diff
@@ -344,6 +344,8 @@ void ThreadPool::shutdown() {
     for (auto* t : _tokens) {
         DCHECK(t->state() == ThreadPoolToken::State::IDLE || t->state() == ThreadPoolToken::State::QUIESCED);
     }
+    // releasing the tasks outside of lock
+    l.unlock();
```

Comment:
why need an unlock explicitly.

Reply:
The tasks in the queue `to_release` will be destructed which should be executed out of the lock

Reply:
maybe move the `to_release` definition before the unique_lock and add comment to explain that the order of the variables definition matters.

Reply:
fixed. also refactor ThreadPoolToken::shutdown